client/LaunchVM.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2022 IBM
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import base64
+import libvirt
+import shutil
+import grpc
+import hashlib
+from uuid import UUID
+from bs4 import BeautifulSoup as bs
+
+from keybroker_pb2_grpc import KeyBrokerServiceStub
+from keybroker_pb2 import BundleRequest, SecretRequest, RequestDetails
+
+XML_PATH = "sev_guest.xml"
+KBS_URI = "127.0.0.1:44444"
+
+def main():
+    hv = libvirt.open("qemu:///system")
+
+    # load and parse XML
+    f = open(XML_PATH, "r")
+    xml = f.read()
+    f.close()
+
+    soup = bs(xml, "xml")
+    
+    policy = soup.find("policy").text
+
+    # use this to calculate the expected digest
+    ovmf_path = soup.find("loader").text
+    initrd_path = soup.find("initrd").text
+    kernel_path = soup.find("kernel").text
+    cmdline = soup.find("cmdline").text
+
+    channel = grpc.insecure_channel(KBS_URI)
+    client = KeyBrokerServiceStub(channel)
+   
+    
+    # currrently libvirt cannot export a cert chain with a 
+    # signed CEK. instead, use sevctl to export the cert chain
+    # and store it in a file.
+    f = open("certchain","rb")
+    cert_chain_bytes = f.read()
+    f.close()
+    cert_chain = base64.b64encode(cert_chain_bytes).decode("utf-8")
```

Comment:
Are you sure `utf-8` is a valid encoding name?

```
>>> from encodings.aliases import aliases
>>> def find(q):
...     return [(k,v) for k, v in aliases.items() if q in k or q in v]
... 
>>> find ('utf8')
[('utf8_ucs4', 'utf_8'), ('utf8_ucs2', 'utf_8'), ('utf8', 'utf_8')]
>>> find ("utf8")
[('utf8_ucs4', 'utf_8'), ('utf8_ucs2', 'utf_8'), ('utf8', 'utf_8')]
>>> find ("utf-8")
[]
```

Or maybe that's using a different alias list?

Reply:
OK. So the codecs library is not consistent with the aliases. One more python weirdness to remember, I guess.